paper-dialog-behavior.html

```diff
@@ -140,12 +140,13 @@
       } else {
         this.setAttribute('aria-modal', 'false');
       }
-      // modal implies noCancelOnOutsideClick and withBackdrop if true, don't overwrite
-      // those properties otherwise.
```

Comment:
@cdata I've extended the tests and noticed it was possible to reproduce the issue. Now I'm storing the old values, and resetting them when modal is set to `false` (if their current value is still true). A bit tricky to explain by words, I think the tests are more explicative :) PTAL